internal/registration/internal/domain/user_registration.go

```diff
@@ -0,0 +1,70 @@
+package domain
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"time"
+)
+
+var (
+	UserRegistrationCannotBeConfirmedMoreThanOnceError = fmt.Errorf("UserRegistration can't be confirmed more than once")
+)
+
+func RegisterNewUser(firstName, lastName, email, password string) *UserRegistration {
+	return &UserRegistration{
+		ID:               UserRegistrationId{Value: uuid.New()},
```

Comment:
You could create a function that creates UserRegistrationID.
Also you could create a separate function that parse UserRegistrationID.
uuid.UUID{} is not a valid value for UserRegistrationID.